lib/checker.js

```diff
@@ -60,15 +93,22 @@ Checker.prototype.checkDirectory = function(path) {
             }
 
             return vowFs.stat(fullname).then(function(stat) {
+                var promise;
+
                 if (stat.isDirectory()) {
                     return this.checkDirectory(fullname);
                 }
 
                 if (!this._hasCorrectExtension(fullname)) {
-                    return [];
+                    if (this._configuration.getExtract() === 'never') {
+                        return [];
+                    }
+                    promise = Vow.when(this.extractFile(fullname));
```

Comment:
Maybe add support for array of minimatch masks in additional of `never`, `auto` and `always`? And set `['*.html', '*.htm', '*.xhtml']` or `['*.(html?|xhtml)']` by default?

```
{
  "extract": ['*.html', '*.htm', '*.xhtml']
}
```

```
jscs . --extract *.html *.xhtml
```